router/network.go

```diff
@@ -182,11 +184,12 @@ func (network *NetHandleT) SendPost(ctx context.Context, structData integrations
 			//Detecting content type of the respBody
 			contentTypeHeader = http.DetectContentType(respBody)
 		}
+		mediaType, _, _ := mime.ParseMediaType(contentTypeHeader)
 
-		// If content type is not of type "*text*", overriding it with empty string
-		if !(strings.Contains(strings.ToLower(contentTypeHeader), "text") ||
-			strings.Contains(strings.ToLower(contentTypeHeader), "application/json") ||
-			strings.Contains(strings.ToLower(contentTypeHeader), "application/xml")) {
+		// If media type is not in some human readable format (text,json,xml), override the response with an empty string
+		// https://www.iana.org/assignments/media-types/media-types.xhtml
+		isHumanReadable, _ := regexp.MatchString(`^(text/[a-z0-9.-]+)|(application/([a-z0-9.-]+\+)?(json|xml))$`, mediaType)
```

Comment:
For performance reasons, I would suggest that you [compile](https://pkg.go.dev/regexp#MustCompile) once and then use it here.

You can do that by using init() and compiling as a global variable

Reply:
Also, I have to say I am not very fun of using regex. They are hard to read,  to ensure correctness, and can impact performance.

In this case, It seems to be better than doing multiple Contains, and the json / xml suffix would make the existing approach even more complex.

It would be nice, to add some tests for this. Maybe Isolate this to a separate function. Some test cases would help document this function.

Reply:
> For performance reasons, I would suggest that you [compile](https://pkg.go.dev/regexp#MustCompile) once and then use it here.
> 
> You can do that by using init() and compiling as a global variable

would this
```
var contentTypeRegex *regexp.Regexp

func init() {
	contentTypeRegex = regexp.MustCompile(`^(text/[a-z0-9.-]+)|(application/([a-z0-9.-]+\+)?(json|xml))$`)
}
```
be better than that?
```
var contentTypeRegex = regexp.MustCompile(`^(text/[a-z0-9.-]+)|(application/([a-z0-9.-]+\+)?(json|xml))$`)
```

why?